common/logisticspipes/network/GuiHandler.java

```diff
@@ -261,6 +262,10 @@ public Object getClientGuiElement(int ID, EntityPlayer player, final World world
 			return getClientGuiElement(-100 * 20 + x, player, world, 0, -1, z);
 		}
 
+		if (ID == GuiIDs.GUI_Guide_Book_ID) {
+			return new GuiGuideBook();
```

Comment:
Hmm, don't do the GUI construction here, you won't be able to pass the additional parameters that you will need. You could use the x,y,z coordinates for extra data if you really want to, though, instead of the player's coordinates, since they're not actually used for anything.

Reply:
Where should I make it? I thought I had to do it here

Reply:
I'd add a method to the client/common proxy (don't construct it directly from the item class) that opens the gui (obviously only on the client side, on the server it's a no-op), and call that from the onItemUse method instead of this one

Reply:
I couldn't make this elsewhere, but as I read that the x y z are not needed to be actual coordinates but could be any arbitrary value I am using the x value as a way to pass the hand through to the Gui.
Until I find a proper solution I think this will have to be :(